src/utils/bwellPersonFinder.js

```diff
@@ -33,12 +36,247 @@ class BwellPersonFinder {
         });
 
         return await this.searchForBwellPersonAsync({
-            currentSubject: `Patient/${patientId}`,
+            currentSubject: `${PATIENT_REFERENCE_PREFIX}${patientId}`,
             databaseQueryManager: databaseQueryManager,
             visitedSubjects: new Set()
         });
     }
 
+    /**
+     * finds bwell person Ids associated with patientsIds
+     * @param {string[]} patientIds List of patient Ids
+     * @returns {Promise<Map<string, string>>} Returns map with key as patientId and value as master-persons-id
+     */
+    async getBwellPersonIdsAsync({
+        patientIds
+    }) {
+        /**@type {string[]} */
+        const patientReferences = [];
+        /**@type {string[]} */
+        const proxyPatientIds = new Set();
+
+        patientIds.forEach((id) => {
+            if (id.startsWith(`${PERSON_PROXY_PREFIX}`)) {
+                proxyPatientIds.add(id);
+            } else {
+                patientReferences.push(`${PATIENT_REFERENCE_PREFIX}${id}`);
+            }
+        });
+        const databaseQueryManager = this.databaseQueryFactory.createQuery({
+            resourceType: 'Person',
+            base_version: '4_0_0'
+        });
+
+        const patientsToBwellPerson = await this.searchForBwellPersonsAsync({
+            currentReferences: patientReferences,
+            databaseQueryManager,
+            level: 0,
+            visitedReferences: new Set(),
+        });
+
+        /**
+         * check if proxy patient is bwell master person,
+         * it must be available in patientsToBwellPerson Map
+         */
+        for (const /**@type {string} */ masterPerson of patientsToBwellPerson.values()) {
+            const personID = masterPerson.replace(`${PERSON_REFERENCE_PREFIX}`, '');
+            const proxyPatient = `${PERSON_PROXY_PREFIX}${personID}`;
+            if (proxyPatientIds.has(proxyPatient)) {
+                patientsToBwellPerson.set(proxyPatient, masterPerson);
+                proxyPatientIds.delete(proxyPatient);
+            }
+        }
+        // Process remaining proxy patient, it can be non master person ID only
+        const proxyPatientsToBwellPerson = await this.searchForBwellPersonsAsync({
+            currentReferences: [...proxyPatientIds].map((proxyPatent) => `${PERSON_REFERENCE_PREFIX}${proxyPatent.replace(`${PERSON_PROXY_PREFIX}`, '')}`),
```

Comment:
Any need to create a new array for `proxyPatientIds` here? `.map()` will also return a new array anyway.